### PR TITLE
zoxide: init at 0.2.0

### PR DESCRIPTION
**Pull request by ysndr**
###### Motivation for this change
I started using zoxide as an alternative to `z` and thought it would be nice to have it included in nixpkgs as well.
zoxide seems to be in a very early state though so I'd be fine to include it in my personal nur repo for some time until it proves stable enough to be added here.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
Might want to wait until v0.2.0 is released, which will remove the fish/zsh/bash init scripts (in your `preFixup`) in favor of evaluating the output of a subcommand. It's currently over on the v0.2.0 branch for now.

**Comment by ysndr** [comment]
Sure no need to rush that

**Comment by ajeetdsouza** [comment]
zoxide v0.2.0 has been released!

**Comment by ysndr** [comment]
Updated to 0.2.0

File: pkgs/tools/misc/zoxide/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ajeetdsouza";
+    repo = "zoxide";
+    rev = "v${version}";
+    sha256 = "0s0zrkn48rgsf2fqpgsl0kix4l6w7k7pwssvdja6y3a4c097qmnm";
+  };
+
+  buildInputs = [
+    # Example Run-time Additional Dependencies
```
Comment (cole-h) [inline pkgs/tools/misc/zoxide/default.nix]:
```suggestion
```

**Review by cole-h — CHANGES_REQUESTED** [review]
A minor suggestion, then it's g2g!

**Comment by ysndr** [comment]
Changed, thanks

**Review by cole-h — APPROVED** [review]
LGTM. Binary runs as expected. Thanks!

```
[4 built, 2 copied (0.1 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/82282
1 package built:
zoxide
```

**Pull request merged**